meshing/src/CalculateNormals.cpp

```diff
@@ -22,6 +22,28 @@ void CalculateNormals::compute_normals_()
 {
     vertexNormals_ = std::vector<cv::Vec3d>(output_->GetNumberOfPoints(), 0);
 
+    // Compute the average mesh coordinates, to use as reference when choosing
+    // an orientation for the mesh normals. Should produce consistent result on
+    // convex-ish meshes.
+    unsigned int pointCount = 0;
+    ITKPoint pointAverage;
+    pointAverage.Fill(0.0);
+    for (auto it = input_->GetPoints()->Begin(); it != input_->GetPoints()->End(); ++it) {
+        auto point = it.Value();
+        pointAverage[0] += point[0];
+        pointAverage[1] += point[1];
+        pointAverage[2] += point[2];
+        pointCount++;
+    }
+    if (pointCount > 0) {
+        pointAverage[0] /= pointCount;
+        pointAverage[1] /= pointCount;
+        pointAverage[2] /= pointCount;
+    }
```

Comment:
Yeah could be, I don't know if it is really an issue. Another option (which also saves on compute) is to sample some of the points instead of using all of them for the average.

Reply:
Also reasonable and keeps it roughly constant time. I'm fine with either approach.